receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -257,6 +312,7 @@ func gcInterval(cfg *config.Config) time.Duration {
 func (r *pReceiver) Shutdown(context.Context) error {
 	r.cancelFunc()
 	r.scrapeManager.Stop()
+	r.reloadReady.Close()
```

Comment:
I don't think this is necessary.  You don't need to close channels on shutdown for them to be GC'd (this would be neccessary if closing the channel causes things waiting on the channel to shut down.